packages/flame_forge2d/example/lib/widget_sample.dart

```diff
@@ -0,0 +1,124 @@
+import 'package:flame/components.dart';
+import 'package:flame/game.dart';
+import 'package:flame/input.dart';
+import 'package:flame_forge2d/flame_forge2d.dart';
+import 'package:flame_forge2d/forge2d_game.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/widgets.dart' as widgets;
+import 'package:forge2d/forge2d.dart' hide Transform;
+
+import 'boundaries.dart';
+
+class WidgetSample extends Forge2DGame with TapDetector {
+  List<Function()> updateStates = [];
+  Map<int, Body> bodyIdMap = {};
+  List<int> addLaterIds = [];
+
+  Vector2 screenPosition(Body body) => worldToScreen(body.worldCenter);
+
+  WidgetSample() : super(zoom: 20, gravity: Vector2(0, -10.0));
+
+  @override
+  Future<void> onLoad() async {
+    final boundaries = createBoundaries(this);
+    addAll(boundaries);
+  }
+
+  Body createBody() {
+    final bodyDef = BodyDef()
+      ..angularVelocity = 3
+      ..position =
+          screenToWorld(Vector2.random()..multiply(viewport.effectiveSize))
+      ..type = BodyType.dynamic;
+    final body = world.createBody(bodyDef);
+
+    final shape = PolygonShape()..setAsBoxXY(4.6, 0.8);
+    final fixtureDef = FixtureDef(shape)
+      ..density = 1.0
+      ..restitution = 0.95;
+    body.createFixture(fixtureDef);
+    return body;
+  }
+
+  int createBodyId() {
+    final id = bodyIdMap.length + addLaterIds.length;
+    addLaterIds.add(id);
+    return id;
+  }
+
+  @override
+  void update(double dt) {
+    super.update(dt);
+    addLaterIds.forEach((id) => bodyIdMap[id] = createBody());
+    addLaterIds.clear();
+    updateStates.forEach((f) => f());
+  }
+
+  static GameWidget gameWidget() {
```

Comment:
Methods that returns widgets is something that should be avoided (more on why [here](https://stackoverflow.com/questions/53234825/what-is-the-difference-between-functions-and-classes-to-create-reusable-widgets/53234826#53234826)).

I think instead of this being a static method inside the game class, we should just create a StatelessWidget that builds this.

Reply:
TIL! nice to know

Reply:
Good to know indeed! Fixed now :)